.changeset/happy-falcons-walk.md

```diff
@@ -0,0 +1,5 @@
+---
+'openzeppelin-solidity': major
+---
+
+`TransparentUpgradeableProxy`: Added an immutable admin set during construction to avoid unnecessary storage reads on every proxy call, and removed the ability to change the admin from both proxy and `ProxyAdmin`.
```

Comment:
I think splitting the sentence is clearer. Also, I don't sure what you mean by "remove the ability to change the admin from the ProxyAdmin. The ProxyAdmin is still ownable.

```suggestion
`TransparentUpgradeableProxy`: Admin is now stored in an immutable variable (set during construction) to avoid unnecessary storage reads on every proxy call. This removed the ability to ever change the admin. Transfer of the upgrade capability is exclusively handled though the ownership of the `ProxyAdmin`.
```